scripts/windows/test/Run-E2ETest.ps1

```diff
@@ -620,26 +654,152 @@ Function RunTempSensorTest
     Return $testExitCode
 }
 
+Function RunLeafDeviceTest
+(
+    [ValidateSet("sas","x509CA","x509Thumprint")][string]$authType,
+    [ValidateSet("Mqtt","MqttWs","Amqp", "AmqpWs")][string]$protocol,
+    [ValidateNotNullOrEmpty()][string]$leafDeviceId,
+    [string]$edgeDeviceId=$NULL
```

Comment:
need to set to $NULL?

Reply:
I wanted to make this an optional arg that's why I set it as such.